### PR TITLE
1.3 apis tools rename

### PR DESCRIPTION
**Pull request by christinaausley**
## What is the purpose of the change

APIs/Clients -> APIs/Tools rename for 1.3 docs.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: versioned_docs/version-1.2/self-managed/zeebe-deployment/index.md
Comment (akeller) [inline versioned_docs/version-1.2/self-managed/zeebe-deployment/index.md:1]: Looks like this 1.2 page snuck in.

**Comment by christinaausley** [comment]
@pepopowitz @akeller This one is also good to go 👍

**Review by akeller — COMMENTED** [review]
Building this locally looks (how adorable was 1.3 🥺) and works like I expect, however I want to mention the following:

* The top nav still says "APIs & Clients."
  * I assume that's intended in this PR.
* The sidebar and overview page title still say APIs & Clients
  * Feels like this should be changed in this PR(?)
 
The URL is http://localhost:3001/docs/1.3/apis-tools/working-with-apis-tools/, as intended.

~Also looks like one of the 1.2 pages snuck in. Not a huge deal, but figured I would mention it.~

**Comment by akeller** [comment]
@christinaausley I was too fast!! My bad.

File: versioned_docs/version-1.2/self-managed/zeebe-deployment/index.md
Comment (akeller) [reply versioned_docs/version-1.2/self-managed/zeebe-deployment/index.md:1]: Poof gone!

**Review by pepopowitz — APPROVED** [review]
:shipit:

**Review by akeller — APPROVED** [review]
:shipit:

**Pull request merged**